modules/Kconfig.ambiq

```diff
@@ -0,0 +1,19 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2023 Antmicro <www.antmicro.com>
+
+config HAS_AMBIQ_HAL
```

Comment:
`HAS_` is really reserved to indicate certain features are available (e.g. `HAS_FPU`), even though this has been abused by some HALs. Selectable symbols should also be promptless.

What I'd suggest here is to:

1. rename to `AMBIQ_HAL` (not promptless, as `HAS_` should be)
2. Drivers (dt-based, so can use select) do: `select AMBIQ_HAL` and `select AMBIQ_HAL_USE_XXX`.

Reply:
Ok, now I get it. I changed that also.